src/libsyntax/tokenstream.rs

```diff
@@ -225,13 +234,107 @@ impl TokenStream {
         }
         true
     }
+
+    pub fn as_tree(self) -> (TokenTree, bool /* joint? */) {
```

Comment:
Could these public methods get docs please? (In particular here, some more explanation of what joint means would be good).